### PR TITLE
codeintel: Repo-independent upload and index pages

### PR DESCRIPTION
**Pull request by efritz**
This covers https://github.com/sourcegraph/sourcegraph/issues/10837 and https://github.com/sourcegraph/sourcegraph/issues/10835.

Site-admins should be able to see LSIF uploads and auto-index behavior without having to look at a particular repository. This is especially true for auto-indexing as we currently don't have a way to determine what is being indexed without (1) blindly looking at repository settings pages, or (2) having direct access to postgres and some knowledge about the auto-indexing heuristic (which will be rapidly changing).

This simply makes the code intelligence repo settings pages available from the site-admin page and drops the repository filter from the queries.

Additional UX improvements will be done in https://github.com/sourcegraph/sourcegraph/pull/11320

### CODE REVIEW

File: web/src/enterprise/codeintel/backend.tsx [hunk 1]
```diff
@@ -1,24 +1,73 @@
...
+} from '../../../../shared/src/graphql/graphql'
 import { map } from 'rxjs/operators'
 import { Observable } from 'rxjs'
-import { queryGraphQL, mutateGraphQL } from '../../../backend/graphql'
+import { queryGraphQL, mutateGraphQL } from '../../backend/graphql'
 
-/**
- * Fetch LSIF uploads for a repository.
- */
 export function fetchLsifUploads({
```
Comment (attfarhan) [inline web/src/enterprise/codeintel/backend.tsx:11]: Can we get a docblock here? In the case when repository is undefined, are we just fetching all lsifUploads?

Comment (attfarhan) [inline web/src/enterprise/codeintel/backend.tsx:11]: Can we add a docblock here? When `repository` is undefined, we're just fetching _all_ LSIF uploads?

File: web/src/enterprise/codeintel/backend.tsx [hunk 2]
```diff
@@ -164,16 +207,51 @@ export function deleteLsifUpload({ id }: { id: string }): Observable<void> {
     )
 }
 
-/**
- * Fetch LSIF indexes for a repository.
- */
 export function fetchLsifIndexes({
```
Comment (attfarhan) [inline web/src/enterprise/codeintel/backend.tsx:210]: Can we add a docblock here?

Comment (efritz) [reply web/src/enterprise/codeintel/backend.tsx:218]: Done.

File: web/src/enterprise/codeintel/backend.tsx (hunk 1 above)
Comment (efritz) [reply web/src/enterprise/codeintel/backend.tsx:15]: Done.

**Comment by unknwon** [comment]
>Site-admins should be able to see LSIF uploads and auto-index behavior without having to look at a particular repository.

This sounds very interesting! Any screenshots or an URL from sourcegraph.com that can view the current UI?

**Comment by efritz** [comment]
@unknwon its the same page you see from repo settings, but I have another draft PR that is trying to improve the layout and UX

**Pull request merged**